### PR TITLE
create #530_Minimum_Absolute_Difference_in_BST.py

### LINKED ISSUE #79: #530_Minimum_Absolute_Difference_in_BST
# Summary
add 530. Minimum Absolute Difference in BST

# Goal
add 530. Minimum Absolute Difference in BST

# Todo
- [ ] add 530. Minimum Absolute Difference in BST
- [ ] time complexity: O(n)
- [ ] space complexity: O(1)

# Deadline
/

# Parent issue
None

# References
None

# Notes
None

### PR DESCRIPTION
**Pull request by ashizawaN**
Closes #79 

Please check my code. 
I solved it with only one solution because I did not think there was another solution that would change the complexity.

### CODE REVIEW

File: #530_Minimum_Absolute_Difference_in_BST.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# time complexity: O(n)
+# space complexity: O(1)
```
Comment (nayuta-ai) [inline #530_Minimum_Absolute_Difference_in_BST.py:2]: The space complexity is O(log n).
A value returned from a recursion method is saved in memory, so the space complexity is O(the height of the recursion). $2^h = n <=> h = log n$.

File: #530_Minimum_Absolute_Difference_in_BST.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+from math import inf
+
+
+class TreeNode:
+    def __init__(self, val=0, left=None, right=None):
```
Comment (nayuta-ai) [inline #530_Minimum_Absolute_Difference_in_BST.py:7]: Write a type annotation.
When an argument use itself type, represent itself type, adding double quotations.
When arguments of a TreeNode have None object, use an Optional class.
```
from typing import Optional

def __init__(self, val: int = 0, left: Optional["TreeNode"] = None, right: Optional["TreeNode"] = None) -> None:
```
[Ref](https://docs.python.org/3/library/typing.html#typing.Optional)

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

File: #530_Minimum_Absolute_Difference_in_BST.py (hunk 1 above)
Comment (ashizawaN) [reply #530_Minimum_Absolute_Difference_in_BST.py:2]: Thank you for your indication. I've understood it and revised it.

File: #530_Minimum_Absolute_Difference_in_BST.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class Solution:
+    def getMinimumDifference(self, root: TreeNode) -> int:
+        def rec(node: TreeNode, low: int, high: int) -> int:
+            if not node:
+                return high - low
+            left = rec(node.left, low, node.val)
+            right = rec(node.right, node.val, high)
+            return min(left, right)
```
[the remark this answers is not in the document]
Comment (nayuta-ai) [reply #530_Minimum_Absolute_Difference_in_BST.py:22]: Add a blank line after a return statement of an inner function.

**Review by nayuta-ai — CHANGES_REQUESTED** [review]

**Review by nayuta-ai — APPROVED** [review]

**Pull request merged**